src/chainstatemanager_opts.h

```diff
@@ -17,6 +17,7 @@ class CChainParams;
  */
 struct ChainstateManagerOpts {
     const CChainParams& chainparams;
+    const std::function<int64_t()> adjusted_time_callback{nullptr};
```

Comment:
good candidate for a not_null https://github.com/bitcoin/bitcoin/issues/24423

given that we assert not null where it is actually used.